src/Libraries/Nop.Data/Migrations/UpgradeTo460/DataMigration.cs

```diff
@@ -0,0 +1,169 @@
+            var languageIdCustomerColumnName = "LanguageId";
+            var taxDisplayTypeIdCustomerColumnName = "TaxDisplayTypeId";
+            var euCookieLawAcceptedCustomerColumnName = "EuCookieLawAccepted";
+
+            if (!Schema.Table(customerTableName).Column(firstNameCustomerColumnName).Exists())
+            {
+                Alter.Table(customerTableName)
+                    .AddColumn(firstNameCustomerColumnName).AsString().Nullable();
+            }
+            if (!Schema.Table(customerTableName).Column(lastNameCustomerColumnName).Exists())
+            {
+                Alter.Table(customerTableName)
+                    .AddColumn(lastNameCustomerColumnName).AsString().Nullable();
+            }
+            if (!Schema.Table(customerTableName).Column(genderCustomerColumnName).Exists())
+            {
+                Alter.Table(customerTableName)
+                    .AddColumn(genderCustomerColumnName).AsString().Nullable();
+            }
+            if (!Schema.Table(customerTableName).Column(dobCustomerColumnName).Exists())
+            {
+                Alter.Table(customerTableName)
+                    .AddColumn(dobCustomerColumnName).AsDateTime().Nullable();
+            }
+            if (!Schema.Table(customerTableName).Column(companyCustomerColumnName).Exists())
+            {
+                Alter.Table(customerTableName)
+                    .AddColumn(companyCustomerColumnName).AsString().Nullable();
+            }
+            if (!Schema.Table(customerTableName).Column(address1CustomerColumnName).Exists())
+            {
+                Alter.Table(customerTableName)
+                    .AddColumn(address1CustomerColumnName).AsString().Nullable();
+            }
+            if (!Schema.Table(customerTableName).Column(address2CustomerColumnName).Exists())
+            {
+                Alter.Table(customerTableName)
+                    .AddColumn(address2CustomerColumnName).AsString().Nullable();
+            }
+            if (!Schema.Table(customerTableName).Column(zioCustomerColumnName).Exists())
+            {
+                Alter.Table(customerTableName)
+                    .AddColumn(zioCustomerColumnName).AsString().Nullable();
+            }
+            if (!Schema.Table(customerTableName).Column(cityCustomerColumnName).Exists())
+            {
+                Alter.Table(customerTableName)
+                    .AddColumn(cityCustomerColumnName).AsString().Nullable();
+            }
+            if (!Schema.Table(customerTableName).Column(countyCustomerColumnName).Exists())
+            {
+                Alter.Table(customerTableName)
+                    .AddColumn(countyCustomerColumnName).AsString().Nullable();
+            }
+            if (!Schema.Table(customerTableName).Column(countryIdCustomerColumnName).Exists())
+            {
+                Alter.Table(customerTableName)
+                    .AddColumn(countryIdCustomerColumnName).AsInt32().WithDefaultValue(0);
```

Comment:
`NotNullable().SetExistingRowsTo()` should be used instead of `WithDefaultValue()`. See sample [here](https://github.com/nopSolutions/nopCommerce/blob/master/src/Libraries/Nop.Data/Migrations/UpgradeTo440/DataMigration.cs#L134).

Reply:
@mhsjaber Not only in this line, but other places also need to be changed.